src/components/Layout/Header/NavBar/ChainMenu.tsx

```diff
@@ -0,0 +1,125 @@
+import { ChevronDownIcon } from '@chakra-ui/icons'
+import { Menu, MenuButton, MenuGroup, MenuItem, MenuList } from '@chakra-ui/menu'
+import { Box, Button, Flex, Text, useColorModeValue } from '@chakra-ui/react'
+import { CHAIN_NAMESPACE, ChainId, fromChainId } from '@shapeshiftoss/caip'
+import { supportsEthSwitchChain } from '@shapeshiftoss/hdwallet-core'
+import { bnOrZero } from '@shapeshiftoss/investor-foxy'
+import { useEffect, useMemo, useState } from 'react'
+import { AssetIcon } from 'components/AssetIcon'
+import { CircleIcon } from 'components/Icons/Circle'
+import { getChainAdapters } from 'context/PluginProvider/PluginProvider'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { selectAssetById, selectFeatureFlags } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+const ChainMenuItem: React.FC<{
+  chainId: ChainId
+  onClick: (chainId: ChainId) => void
+  isConnected: boolean
+}> = ({ chainId, onClick, isConnected }) => {
+  const chainName = getChainAdapters().get(chainId)?.getDisplayName()
+  const { chainReference: evmChainId } = fromChainId(chainId)
+  const nativeAssetId = getChainAdapters().get(chainId)?.getFeeAssetId()
+  const nativeAsset = useAppSelector(state => selectAssetById(state, nativeAssetId ?? ''))
+
+  const connectedIconColor = useColorModeValue('green.500', 'green.200')
+  const connectedChainBgColor = useColorModeValue('blackAlpha.100', 'whiteAlpha.50')
+
+  if (!nativeAsset) return null
+
+  return (
+    <MenuItem
+      icon={<AssetIcon src={nativeAsset.icon ?? ''} width='6' height='auto' />}
```

Comment:
icon is required unless I am missing something
```suggestion
      icon={<AssetIcon src={nativeAsset.icon} width='6' height='auto' />}
```